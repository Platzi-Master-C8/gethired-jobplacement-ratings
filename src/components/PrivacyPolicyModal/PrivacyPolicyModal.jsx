/* eslint-disable jsx-a11y/anchor-is-valid */
import React, { useState } from 'react';
import { Box, Grid, Link, Modal, Typography, Button } from '@mui/material';

import { useMediaQueryContext } from '../../context/MediaQueryContext';

const boxStyles = (isMobile) => ({
    position: 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
    width: isMobile.medium ? '75%' : '65%',
    height: '75%',
    bgcolor: 'white',
    boxShadow: 24,
    p: 4,
    overflow: 'auto',
});

const PrivacyPolicyModal = () => {
    const [open, setOpen] = useState(false);

    const isMobile = useMediaQueryContext();

    return (
        <div>
            <Grid item md={12} sx={{ display: 'grid', justifyContent: 'flex-end' }}>
                <Link
                    underline="none"
                    variant="caption"
                    sx={{ cursor: 'pointer', wordWrap: 'break-word', whiteSpace: 'pre-wrap', mx: 1 }}
                    onClick={() => setOpen(true)}
                >
                    Politica de Privacidad
                </Link>
            </Grid>
            <Modal
                open={open}
                onClose={() => setOpen(false)}
                aria-labelledby="modal-modal-title"
                aria-describedby="modal-modal-description"
            >
                <Box sx={boxStyles(isMobile)}>
                    <Typography variant="h2" sx={{ textAlign: 'center', mb: 2 }}>
                        Politica de Privacidad
                    </Typography>
                    <Typography variant="text">
                        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut
                        labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco
                        laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in
                        voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat
                        non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                    </Typography>
                    <Typography variant="text">
                        Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque
                        laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto
                        beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut
                        odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt.
                        Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit,
                        sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat
                        voluptatem. Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit
                        laboriosam, nisi ut aliquid ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui
                        in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat
                        quo voluptas nulla pariatur?
                    </Typography>
                    <Typography variant="text">
                        At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum
                        deleniti atque corrupti quos dolores et quas molestias excepturi sint occaecati cupiditate non
                        provident, similique sunt in culpa qui officia deserunt mollitia animi, id est laborum et
                        dolorum fuga. Et harum quidem rerum facilis est et expedita distinctio. Nam libero tempore, cum
                        soluta nobis est eligendi optio cumque nihil impedit quo minus id quod maxime placeat facere
                        possimus, omnis voluptas assumenda est, omnis dolor repellendus. Temporibus autem quibusdam et
                        aut officiis debitis aut rerum necessitatibus saepe eveniet ut et voluptates repudiandae sint et
                        molestiae non recusandae. Itaque earum rerum hic tenetur a sapiente delectus, ut aut reiciendis
                        voluptatibus maiores alias consequatur aut perferendis doloribus asperiores repellat.
                    </Typography>
                    <Grid container sx={{ alignItems: 'center', justifyContent: 'center', marginTop: 4 }}>
                        <Button variant="contained" onClick={() => setOpen(false)}>
                            Return
                        </Button>
                    </Grid>
                </Box>
            </Modal>
        </div>
    );
};

export default PrivacyPolicyModal;
