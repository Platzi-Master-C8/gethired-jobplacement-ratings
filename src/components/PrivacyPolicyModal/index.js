import PrivacyPolicyModal from './PrivacyPolicyModal';

export { PrivacyPolicyModal };
