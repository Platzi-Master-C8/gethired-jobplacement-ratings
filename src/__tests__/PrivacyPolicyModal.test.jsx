import React from 'react';

import { render } from '@testing-library/react';
import { PrivacyPolicyModal } from 'Components/PrivacyPolicyModal';

describe('Component <PrivacyPolicyModal/>', () => {
    test('renders PrivacyPolicyModal', () => {
        const { asFragment } = render(<PrivacyPolicyModal />);
        expect(asFragment()).toMatchSnapshot();
    });
});
