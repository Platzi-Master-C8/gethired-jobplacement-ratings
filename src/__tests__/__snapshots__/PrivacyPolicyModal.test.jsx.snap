// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <PrivacyPolicyModal/> renders PrivacyPolicyModal 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-12 css-oav0mg-MuiGrid-root"
    >
      <a
        class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-v181wt-MuiTypography-root-MuiLink-root"
      >
        Politica de Privacidad
      </a>
    </div>
  </div>
</DocumentFragment>
`;
